tests/test_error_quality.py

```
@@ -229,8 +235,153 @@ def test_inner_missing_field_failure_carries_operation_index(self, doc_path):
+    search scope. `AmbiguousTextError` carries `total_occurrences` and
+    names both recovery paths (explicit `occurrence=` or `find_all()`);
+    explicit `occurrence=0` keeps the old first-match behavior per call.
+    """
+
+    AMBIGUOUS = "alpha beta alpha gamma alpha."
+    UNIQUE = "one needle only."
+
+    def _build_doc(self, doc_path: Path) -> Document:
+        return _build_doc_with_paragraphs(doc_path, [self.AMBIGUOUS, self.UNIQUE])
+
+    def _edit_calls(self, doc: Document, paragraph: str):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_edit_calls`

(ANN202)